src/components/plots/TrajectoryAnalysisPlot.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Vega } from 'react-vega';
+
+import { generateData as generateCategoricalEmbeddingData } from 'utils/plotSpecs/generateEmbeddingCategoricalSpec';
+import {
+  generateSpec,
+  generateData as generateTrajectoryPathData,
+} from 'utils/plotSpecs/generateTrajectoryAnalysisGraph';
+import { loadEmbedding } from 'redux/actions/embedding';
+import { loadCellSets } from 'redux/actions/cellSets';
+import { loadProcessingSettings } from 'redux/actions/experimentSettings';
+
+import { getCellSets } from 'redux/selectors';
+import PlatformError from 'components/PlatformError';
+import Loader from 'components/Loader';
+import changeEmbeddingAxesIfNecessary from 'components/plots/helpers/changeEmbeddingAxesIfNecessary';
+
+const TrajectoryAnalysisPlot = (props) => {
+  const {
+    experimentId,
+    config,
+    plotData,
+    actions,
+    onUpdate,
+  } = props;
+  const dispatch = useDispatch();
+
+  const cellSets = useSelector(getCellSets());
+
+  const embeddingSettings = useSelector(
+    (state) => state.experimentSettings.originalProcessing?.configureEmbedding?.embeddingSettings,
+  );
+
+  const {
+    data: embeddingData,
+    loading: embeddingLoading,
+    error: embeddingError,
+  } = useSelector(
+    (state) => state.embeddings[embeddingSettings?.method],
+  ) || {};
+
+  const [plotSpec, setPlotSpec] = useState({});
+
+  useEffect(() => {
+    if (!embeddingSettings) {
+      dispatch(loadProcessingSettings(experimentId));
+    }
+
+    if (cellSets.loading && !cellSets.error) {
+      dispatch(loadCellSets(experimentId));
+    }
```

Comment:
Why does `loadCellSets` depend on `experimentId` (which can't really change, right?) and also why does it depend on `embeddingSettings?.method`?

Reply:
Good point. Should be able to split up this block